seismiqb/src/crop_batch.py

```diff
@@ -463,83 +333,70 @@ def create_masks(self, ix, dst, src='slices', mode='horizon', width=3, src_label
+        """ Remove elements with masks lesser than a threshold.
 
         Parameters
         ----------
-        src_masks : str
-            component of batch that stores masks.
-        src_slices : str
-            component of batch that stores slices of crops.
-        dst : str
-            component of batch to store the resulting labels.
         threshold : float
-            parameter of mask-thresholding.
-        averaging : str
-            method of pandas.groupby used for finding the center of a horizon.
-        coordinates : str
-            coordinates-mode to use for keys of point-cloud. Can be either 'cubic'
-            or 'lines'. In case of `lines`-option, `geometries` must be loaded as
-            a component of batch.
-        to_numba : bool
-            whether to convert the resulting point-cloud to numba-dict. The conversion
-            takes additional time.
-
-        Returns
-        -------
-        SeismicCropBatch
-            batch with fetched labels.
+            Minimum percentage of covered area (spatial-wise) for a mask to be kept in the batch.
+        passdown : sequence of str
+            Components to filter.
         """
-        _ = dst, to_numba
+        _ = threshold, passdown
+        pos = self.get_pos(None, src, ix)
+        mask = getattr(self, src)[pos]
 
-        # threshold the mask
-        mask = getattr(self, src_masks)[self.get_pos(None, src_masks, ix)]
+        reduced = np.max(mask, axis=-1) > 0.0
+        return np.sum(reduced) / np.prod(reduced.shape)
 
-        # prepare args
-        i_shift, x_shift, h_shift = [self.get(ix, src_slices)[k][0] for k in range(3)]
-        geom = self.get(ix, 'geometries')
-        if coordinates == 'lines':
-            transforms = (lambda i_: geom.ilines[i_ + i_shift], lambda x_: geom.xlines[x_ + x_shift],
-                          lambda h_: h_ + h_shift)
-        else:
-            transforms = (lambda i_: i_ + i_shift, lambda x_: x_ + x_shift,
-                          lambda h_: h_ + h_shift)
+    def _post_mask_rebatch(self, areas, *args, src=None, passdown=None, threshold=None, **kwargs):
+        #pylint: disable=protected-access, access-member-before-definition, attribute-defined-outside-init
+        _ = args, kwargs
+        new_index = [self.indices[i] for i, area in enumerate(areas) if area > threshold]
+        new_dict = {idx: self.index._paths[idx] for idx in new_index}
+        self.index = FilesIndex.from_index(index=new_index, paths=new_dict, dirs=False)
 
-        return mask_to_horizon(mask, threshold, averaging, transforms, separate=False)
+        passdown = passdown or []
+        passdown.extend([src, 'slices'])
+        passdown = list(set(passdown))
+
+        for compo in passdown:
+            new_data = [getattr(self, compo)[i] for i, area in enumerate(areas) if area > threshold]
+            self.add_components(compo, np.array(new_data))
```

Comment:
I think it is reasonable to change this line to
```suggestion
            setattr(self, compo, new_data)
```
 or add an updated add_component so this line won't fail.